modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2626,6 +2626,7 @@ def broadcast_apply_full_axis(
         dtypes=None,
         keep_partitioning=True,
         sync_labels=True,
+        pass_axis_lengths_to_partitions=False,
```

Comment:
Why did we refuse from pass_cols_to_partitions? What if we will want to pass rows to partitions in future regardless of `axis` value?

Reply:
> Why did we refuse from pass_cols_to_partitions?

We don't need it anymore since the function is now row-wise

> What if we will want to pass rows to partitions in future regardless of axis value?

It's unlikely we would want that. We don't need column widths/columns if the `axis=1` since the whole column axis is available in the kernel and the same for `axis=0`. What we potentially want to have when applying a full-axis function is some knowledge about an opposite axis. In this PR this "knowledge" was chosen to be partition shapes.

Reply:
Any suggestion on how to make this option more general? Maybe in the form of a callback, to which it will be possible to pass the `PandasDataframe` and return an arbitrary list of futures that remote function will have to accept?

Reply:
Let's leave as is for now.

Reply:
ok